src/Elastic.Apm/Model/Span.cs

```diff
@@ -43,8 +43,14 @@ IApmLogger logger
 			ParentId = parentId;
 			TraceId = traceId;
 
-			// Started spans should be counted only for sampled transactions
-			if (IsSampled) enclosingTransaction.SpanCount.Started++;
+			if (IsSampled)
+			{
+				// Started spans should be counted only for sampled transactions
+				enclosingTransaction.SpanCount.Started++;
+
+				// Spans are sent only for sampled transactions so it's only worth capturing stack trace for sampled spans
+				StackTrace = StacktraceHelper.GenerateApmStackTrace(new StackTrace(true).GetFrames(), _logger, $"Span `{Name}'");
```

Comment:
I added another comment where you removed capturing stacktrace for outgoing HTTP requests. So.. is the plan to capture spans for every span? If yes, then I think we should definitely implement the debug option to make this opt-in, because this will add overhead.

Reply:
I thought it would be better to have uniform approach - if we capture stack trace at span start let's do it for all spans. I agree that we need a configuration setting to control this behavior. I propose to leave it as it's (stack trace for all spans) in this PR and I've opened #307 to follow up.

BTW stack trace for HTTP spans on .NET Full Framework is not very useful because request-started event is sent to `IDiagnosticListener` in the context that is not related to the originating HTTP client call:

    Elastic.Apm.Model.Span in .ctor
    Elastic.Apm.Model.Transaction in StartSpanInternal
    Elastic.Apm.DiagnosticListeners.HttpDiagnosticListenerImplBase`2 in ProcessStartEvent
    Elastic.Apm.DiagnosticListeners.HttpDiagnosticListenerImplBase`2 in OnNext
    System.Diagnostics.DiagnosticListener in Write
    System.Diagnostics.DiagnosticSource in StartActivity
    System.Diagnostics.HttpHandlerDiagnosticListener in RaiseRequestEvent
    System.Diagnostics.HttpHandlerDiagnosticListener+HttpWebRequestArrayList in Add
    System.Net.Connection in StartRequest
    System.Net.Connection in SubmitRequest
    System.Net.ServicePoint in SubmitRequest
    System.Net.HttpWebRequest in SubmitRequest
    System.Net.HttpWebRequest in BeginGetResponse
    System.Net.Http.HttpClientHandler in StartGettingResponse
    System.Net.Http.HttpClientHandler in StartRequest
    System.Threading.Tasks.Task in InnerInvoke
    System.Threading.Tasks.Task in Execute
    System.Threading.Tasks.Task in ExecutionContextCallback
    System.Threading.ExecutionContext in RunInternal
    System.Threading.ExecutionContext in Run
    System.Threading.Tasks.Task in ExecuteWithThreadLocal
    System.Threading.Tasks.Task in ExecuteEntry
    System.Threading.Tasks.Task in System.Threading.IThreadPoolWorkItem.ExecuteWorkItem
    System.Threading.ThreadPoolWorkQueue in Dispatch
    System.Threading._ThreadPoolWaitCallback in PerformWaitCallback

Reply:
> I propose to leave it as it's (stack trace for all spans) in this PR and I've opened #307 to follow up. 

Sure, that's totally fine. #307 seems to be a small task but relatively important because I feel the added overhead is big (it'd be nice to see numbers..), once this gets merged we should follow up on that. I'm happy to take that one.

> BTW stack trace for HTTP spans on .NET Full Framework is not very useful 

I see, well it's unfortunate, but not a new problem. I'd leave it in this PR as it is and focus on the testing part. We can think about that separately. But yeah... probably we should think about that.

Reply:
I moved #307 to ready bucket so we don't forget to do it for 1.0.